pyo3-derive-backend/src/pymethod.rs

```diff
@@ -244,28 +258,48 @@ pub fn impl_wrap_static(cls: &syn::Type, spec: &FnSpec<'_>) -> TokenStream {
     }
 }
 
-/// Generate functiona wrapper (PyCFunction, PyCFunctionWithKeywords)
-pub(crate) fn impl_wrap_getter(cls: &syn::Type, spec: &FnSpec) -> syn::Result<TokenStream> {
-    let takes_py = match &*spec.args {
-        [] => false,
-        [arg] if utils::if_type_is_python(arg.ty) => true,
-        _ => {
-            return Err(syn::Error::new_spanned(
-                spec.args[0].ty,
-                "Getter function can only have one argument of type pyo3::Python!",
-            ));
-        }
-    };
+fn impl_call_getter(spec: &FnSpec) -> syn::Result<TokenStream> {
+    let (py_arg, args) = split_off_python_arg(&spec.args);
+    if !args.is_empty() {
+        return Err(syn::Error::new_spanned(
+            args[0].ty,
+            "Getter function can only have one argument of type pyo3::Python",
+        ));
+    }
 
     let name = &spec.name;
-    let python_name = &spec.python_name;
-
-    let fncall = if takes_py {
+    let fncall = if py_arg.is_some() {
         quote! { _slf.#name(_py) }
     } else {
         quote! { _slf.#name() }
     };
 
+    Ok(fncall)
+}
+
+/// Generate functiona wrapper (PyCFunction, PyCFunctionWithKeywords)
```

Comment:
typo?

Reply:
This comment actually looks very out of date - I'll update it.